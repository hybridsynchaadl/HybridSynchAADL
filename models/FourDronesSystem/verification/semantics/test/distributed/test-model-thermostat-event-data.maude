
set show timing off .

load ../../interpreter-distributed.maude
load ../syntax/module-thermostat-event-data-noparam.maude


mod TEST-THERMOSTAT-SYSTEM is
  including THERMOSTAT-SYSTEM-SYNTAX .
  including MODEL-TRANSITION-SYSTEM .

  op initState : -> Object .
  eq initState = initialize(model) .

  eq #sample = 4 .
  eq #response = 4 .
endm

rew  [100] {initState | 0 | infty} .

--- count the number of reachable states up to time 25
search [,5] {initState | 0 | infty} =>* {none | T:Time | infty} .
