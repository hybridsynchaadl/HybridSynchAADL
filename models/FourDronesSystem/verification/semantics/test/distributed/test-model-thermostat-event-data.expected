==========================================
rewrite [100] in TEST-THERMOSTAT-SYSTEM : {initState | 0 | infty} .
rewrites: 7757
result GlobalSystem: {< top : System | properties : TimingProperties::Period =>
    {{5}},features : none,subcomponents : (< controller : AsyncThread |
    properties : ((TimingProperties::Period => {{5}}) ; (
    HybridSynchAADL::SamplingTime => {{0.0 .. 2.0}}) ;
    HybridSynchAADL::ResponseTime => {{3.0 .. 4.0}}),features : (< on :
    AsyncDataOutPort | properties : none,content : [0.0] # [false],destination
    : (environment .. on) > < off : AsyncDataOutPort | properties : none,
    content : [0.0] # [false],destination : (environment .. off) > < temp :
    AsyncDataInPort | properties : none,content : [0.0] # [false],cache : [
    -1.5874999999999999] # [true],destination : empty >),subcomponents : (<
    lower : Data | properties : DataModel::InitialValue => {{10}},features :
    none,subcomponents : none,connections : empty,value : [1.0e+1] # [true] > <
    upper : Data | properties : DataModel::InitialValue => {{30}},features :
    none,subcomponents : none,connections : empty,value : [3.0e+1] # [true] > <
    curr : Data | properties : DataModel::InitialValue => {{15}},features :
    none,subcomponents : none,connections : empty,value : [-8.96484375e-1] # [
    true] >),connections : empty,varGen : < "top.controller" >,currState :
    init,completeStates : init,variables : x : Real,transitions : ((init -[on
    dispatch]-> state {(v{x} := ((c[curr] + f[temp]) / [[2]])) ; if(v[x] < c[
    lower])(on !([[1]])) (elsif(v[x] > c[upper])off !([[-1]])) end if ; c{curr}
    := v[x]}) ; state -[otherwise]-> init {skip}),loopBound : 0,transBound : 2,
    period : 5,perTimer : 5/2,msgTimer : infty,execTime : 5/2,execTimer : infty
    > < environment : AsyncEnv | properties : ((TimingProperties::Period => {{
    5}}) ; HybridSynchAADL::MaxClockDeviation => {{1.0000000000000001e-1}}),
    features : (< on : AsyncEnvInPort | properties : none,content : [1.0] # [
    true],envCache : null(Real),target : controller,destination : empty > < off
    : AsyncEnvInPort | properties : none,content : [0.0] # [false],envCache :
    null(Real),target : controller,destination : empty > < out :
    AsyncEnvOutPort | properties : none,content : [0.0] # [false],envCache :
    null(Real),target : controller,destination : (controller .. temp) >),
    subcomponents : (< x : Data | properties : DataModel::InitialValue => {{
    0}},features : none,subcomponents : none,connections : empty,value : [
    -1.8125] # [true] > < e : Data | properties : DataModel::InitialValue => {{
    0}},features : none,subcomponents : none,connections : empty,value : [0.0]
    # [true] >),connections : ((x ==> out) ; (on =>> e) ; off =>> e),varGen : <
    "top.environment" >,currMode : state,jumps : ((init -[on]-> state) ; (init
    -[off]-> init) ; (state -[on]-> state) ; state -[off]-> init),flows : ((
    init [x(t)= ((c[x] + c[e]) + (v[t] * [[2.0000000000000001e-1]]))]) ; state
    [x(t)= ((c[x] + c[e]) - (v[t] * [[1.0000000000000001e-1]]))]),sampling :
    controller :(0,2),response : controller :(3,4),period : 5,perTimer : 5/2,
    msgTimer : infty,samplingTimes : controller : (1/4, 3/4, 5/4, 7/4),
    responseTimes : controller : (25/8, 27/8, 29/8, 31/8),envState : [response,
    controller],envTimer : 5/8 >) > | 55/2 | infty}
==========================================
search [, 5] in TEST-THERMOSTAT-SYSTEM : {initState | 0 | infty} =>* {none |
    T:Time | infty} .

No solution.
states: 264  rewrites: 10794
Bye.
